### PR TITLE
Added offscreen cicd & local cache

### LINKED ISSUE #320: [Bug]: CI/CD not running in offscreen mode
### Current Behavior

The `drive` action that evaluates the agent does not run in offscreen mode. This means that the spectator view is shown on the runner.

### Expected Behavior

The runner does not need to show the spectator view.

### How to reproduce the issue

_No response_

### PR DESCRIPTION
**Pull request by JulianTrommer**
# Description

This PR removes the simulation window on the CI/CD pipeline and switches from the GitHub cache to a local cache to avoid the 10GB limit that GitHub Caches have and to skip the up- and download.

Fixes #320 

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)

## Does this PR introduce a breaking change?

No.

## Most important changes

The CI/CD pipeline on the self-hosted runner now runs without a simulator window.
The docker image cache is now saved locally on the self-hosted runner instead of github.

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works (might be obsolete with CI later on)
- [ ] New and existing unit tests pass locally with my changes (might be obsolete with CI later on)

## Summary by CodeRabbit

- **Bug Fixes**
	- Improved error handling in workflows to ensure critical artifacts are validated before proceeding.

- **New Features**
	- Enhanced Docker build caching mechanisms for better performance.
	- Added artifact retention policy for uploaded PR IDs to manage storage effectively.

- **Chores**
	- Updated Docker Compose command to include a new environment variable affecting execution behavior.
	- Modified VSCode settings to adjust Docker Compose command execution context.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -50,8 +53,8 @@ jobs:
           file: ./build/docker/agent/Dockerfile
           load: true
           tags: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME}}:test
-          cache-from: type=gha
-          cache-to: type=gha,mode=max
+          cache-from: type=local,dest=/cache/
+          cache-to: type=local,src=/cache/,mode=max
```
Comment (ll7) [inline .github/workflows/build.yml:57]: Have you considered that the local cache will not be deleted automatically?

https://docs.docker.com/build/ci/github-actions/cache/#local-cache

File: .vscode/settings.json [hunk 2]
```diff
@@ -22,7 +22,7 @@
     "docker.commands.composeUp": [
         {
             "label": "Compose Up",
-            "template": "xhost +local: && USERNAME=$(whoami) USER_UID=$(id -u) USER_GID=$(id -g) ${composeCommand} ${configurationFile} up"
+            "template": "xhost +local: && USERNAME=$(whoami) USER_UID=$(id -u) USER_GID=$(id -g) RENDER_OFFSCREEN= ${composeCommand} ${configurationFile} up"
```
Comment (ll7) [inline .vscode/settings.json:25]: Have you tested, that the space is not ignored, and the following command is used for `RENDER_OFFSCREEN`?

**Review by ll7 — COMMENTED** [review]
Thank you! I did not know about the local cache. However, we might need to empty the local cache?

File: .vscode/settings.json (hunk 2 above)
Comment (JulianTrommer) [reply .vscode/settings.json:25]: Yes, the command still works and the RENDER_OFFSCREEN is ignored.

File: .github/workflows/build.yml (hunk 1 above)
Comment (JulianTrommer) [reply .github/workflows/build.yml:57]: From what I have seen the cache keeps growing, so automatic deletion should be added.
However if it keeps growing do we even need it if a new image is added to the cache on every build?

**Review by ll7 — APPROVED** [review]

**Pull request merged**